### DISCUSSION TITLE
Customize graphql-kotlin logging

### DISCUSSION THREAD

**Discussion by FidasL** [ExpediaGroup/graphql-kotlin#1189] [Q&A] [answered]
Hello.
Please, is there a way how to customize graphql-kotlin-server logs? They contains everything that we need, but in some requests we use binary data. Then in `DEBUG` level, our logs are overfilled with binary data so there is a problem to read from them any useful information. Is there any option to log some graphql variables as `[binary]` or something like that?

**Comment by FidasL** [comment]
Trying to somehow configure it with custom logback filter / messageConverter  implementation. We will see if it will help

**Comment by smyrick** [comment]
Other than for subscriptions, we really don't have any logs except on startup to log the schema. What logs are you interested in? We are using `org.slf4j.LoggerFactory` so you can provide your own implementation and change what level of data is logged. 

The only `debug` level log we have is the following: https://github.com/ExpediaGroup/graphql-kotlin/blob/f20e65bd47d5d9aef599359c62c029d9667833e9/servers/graphql-kotlin-spring-server/src/main/kotlin/com/expediagroup/graphql/server/spring/subscriptions/ApolloSubscriptionProtocolHandler.kt#L64

**Reply by FidasL** [reply]
Debug level is logging incoming graphql query with provided variables. The main issue is that this logger logs a lot of base64 binary data provided in request variables. 

I want to only log for example something like `variables: image=[binary] ` instead of base64 string

**Comment by dariuszkuc** [comment, marked as answer]
This is actually coming from `graphql-java` -> https://github.com/graphql-java/graphql-java/blob/16.x/src/main/java/graphql/GraphQL.java#L432 

You can probably customize your logger to exclude `graphql-java` logs and use something like a custom interceptor instead.
